src/WebJobs.Script.WebHost/Diagnostics/FunctionsVersionLogger.cs

```diff
@@ -0,0 +1,131 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Host.Scale;
+using Microsoft.Azure.WebJobs.Hosting;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+
+namespace Microsoft.Azure.WebJobs.Script.Diagnostics
+{
+    /// <summary>
+    /// Service responsible for logging versions
+    /// </summary>
+    public class FunctionsVersionLogger : IHostedService, IDisposable
+    {
+        private readonly ILogger _logger;
```

Comment:
Shouldn't have to use the logger factory in the constructor

```suggestion
        private readonly ILogger<FunctionsVersionLogger> _logger;
```

Reply:
Done